src/processor.jl

```diff
@@ -227,6 +227,24 @@ end
 get_parent(proc::ThreadProc) = OSProc(proc.owner)
 default_enabled(proc::ThreadProc) = true
 
+"""
+    DaggerKAProc <: Processor
+
+DaggerKAProc is Processor subtype which launches the user's function with KernelAbstractions.jl.  
+"""
+struct DaggerKAProc <: Processor
+    owner::Int
+end
+
+iscompatible(proc::DaggerKAProc, opts, f, args...) = true
+iscompatible_func(proc::DaggerKAProc, opts, f) = true
+iscompatible_arg(proc::DaggerKAProc, opts, x) = true
+move(ctx, from_proc::OSProc, to_proc::DaggerKAProc, x) = x
+move(ctx, from_proc::DaggerKAProc, to_proc::OSProc, x) = x
+execute!(proc::DaggerKAProc, f, args...) = fetch(@kernel f(args...))
```

Comment:
I am sorry for replying late. So, since the processor won't accept a kernel object I passed the kernel function. ``delayed`` accepts only functions so, I omitted the creation of kernel object and rather passed in kernel function. The object which will be created from ``delayed`` will be a Dagger object.

Reply:
@vchuravy is indicating that this isn't valid since `@kernel` generates a kernel with the kernel language, but doesn't actually execute the kernel, or wait on it. What we want here is for `f` to be the function that executes the already-constructed kernel object (see my comment about fixing `delayed` not accepting non-functions), and have that be called and then waited on in `execute!`.

This way, the user writes their kernel with `@kernel` outside of Dagger, and then passes it to Dagger to execute with arguments, and then be waited on by Dagger's scheduler.

Reply:
ok, I get it now.